docker_cli/_shell_cmds.py

```diff
@@ -0,0 +1,369 @@
+            Defaults to True.
+
+        Returns
+        -------
+        Union[List[str], str]
+            The commands.
+        """
+        ...
+
+    @staticmethod
+    @abstractmethod
+    def generate_package_command(
+        target: str, *, stringify: bool = False
+    ) -> Union[List[str], str]:
+        """
+        Returns a command to install a single package.
+
+        Uses the package tool underlying the package manager if one is
+        available. (e.g. dpkg or rpm)
+
+        Parameters
+        ----------
+        target : str
+            The install target.
+        stringify : bool, optional
+            Whether to return a string or list - if true, will return a string.
+            Defaults to False.
+
+        Returns
+        -------
+        Union[List[str], str]
+            The commands.
+        """
+        ...
+
+    @staticmethod
+    @abstractmethod
+    def generate_configure_command(*, stringify: bool = False) -> Union[List[str], str]:
+        """
+        Returns commands to configure and update the package manager.
+
+        Parameters
+        ----------
+        stringify : bool, optional
+            Whether to return a string or list - if true, will return a string.
+            Defaults to False.
+
+        Returns
+        -------
+        Union[List[str], str]
+            The commands.
+        """
+        ...
+
+    @property
+    @abstractmethod
+    def name(self):
+        """
+        The package manager's command name.
+        """
+        ...
+
+    @property
+    @abstractmethod
+    def file_type(self):
+        """
+        The filetype associated with this package manager.
+        """
+        ...
+
+
+class Yum(PackageManager):
+    @staticmethod
+    def generate_install_command(
+        targets: Iterable[str],
+        *,
+        stringify: bool = False,
+        clean: bool = True,
+    ) -> Union[List[str], str]:
+        retval = []
+        retval += ["yum", "install", "-y"]
+        retval.extend(targets)
+        if clean:
+            retval += ["&&", "yum", "clean", "all"]
+            retval += ["&&", "rm", "-rf", "/var/cache/yum"]
+        if stringify:
+            return " ".join(retval)
+        return retval  # pragma: no cover
+
+    @staticmethod
+    def generate_package_command(  # pragma: no cover
+        target: str, *, stringify: bool = False
+    ) -> Union[List[str], str]:
+        retval = ["rpm", "-i", target]
+        if stringify:
+            return " ".join(retval)
+        return retval
+
+    @staticmethod
+    def generate_configure_command(*, stringify: bool = False) -> Union[List[str], str]:
+        retval = []
+        retval += ["yum", "update", "-y"]
+        retval += ["&&", "echo", "'skip_missing_names_on_install=False'"]
+        retval += [">>", "/etc/yum.conf"]
+        retval += ["&&", "rm", "-rf", "/var/cache/yum"]
+        if stringify:
+            return " ".join(retval)
+        return retval  # pragma: no cover
```

Comment:
Seems a bit cleaner and more readable this way

```suggestion
        retval = textwrap.dedent("""
            yum update -y
              && echo 'skip_missing_names_on_install=False' >> /etc/yum.conf
              && rm -rf /var/cache/yum
        """).strip()
        if not stringify:
            return shlex.split(retval)
        return retval  # pragma: no cover
```